vllm/v1/core/mfu_utils.py

```diff
@@ -0,0 +1,408 @@
+    'aten.sub.Tensor': mfu_pointwise,
+    'aten.mul.Tensor': mfu_pointwise,
+    'aten.ge.Scalar': mfu_pointwise,
+    'aten.lt.Scalar': mfu_pointwise,
+    'aten.__and__.Tensor': mfu_pointwise,
+    'aten.__or__.Tensor': mfu_pointwise,
+    'aten.bitwise_not.default': mfu_pointwise,
+    'aten.to.dtype': mfu_pointwise,
+    'aten.div.Tensor': mfu_pointwise,
+    'aten.pow.Tensor_Scalar': scaled_flop_mfu(mfu_pointwise, 8),  # approx
+    'aten.tanh.default': scaled_flop_mfu(mfu_pointwise, 5),
+    'aten.rsqrt.default': scaled_flop_mfu(mfu_pointwise, 10),  # approx
+    'aten.masked_fill_.Scalar': scaled_flop_mfu(mfu_pointwise, 2),
+    'aten.cat.default': mfu_nop,
+    '<built-in function getitem>': mfu_nop,
+    'aten.zeros.default': mfu_nop,
+    'aten.chunk.default': mfu_nop,
+    'aten.pad.default': mfu_nop,
+    'aten.split_with_sizes.default': mfu_nop,
+    'aten.flatten.default': mfu_nop,
+    'aten.flatten.using_ints': mfu_nop,
+    'aten.index_select.default': mfu_nop,
+    'aten.contiguous.default': mfu_nop,
+    'aten.reshape.default': mfu_nop,
+    'aten.slice.default': mfu_nop,
+    'aten.slice.Tensor': mfu_nop,
+    'aten.alias.default': mfu_nop,
+    'aten.view.default': mfu_nop,
+    'aten.unsqueeze.default': mfu_nop,
+    'aten._assert_tensor_metadata.default': mfu_nop,
+}
+
+
+def get_additional_inputs(nodes, model):
+    additional = {}
+    for node in nodes:
+        sd_keys = model.state_dict().keys()
+        if not isinstance(node, str):
+            continue
+        for k in sd_keys:
+            if node in k:
+                t = model.state_dict()[k]
+                additional[k] = TensorMeta(t.shape, t.dtype)
+    if len(additional) == 0:
+        return None
+    return additional
+
+
+def analyze_node(node, model):
+    if node.op == "call_function":
+        output = get_shape_dtype(node, model)
+        inputs = [get_shape_dtype(arg, model) for arg in node.args]
+        op_name = str(node.target)
+        if op_name in mapping:
+            return mapping[op_name](inputs, output)
+        # assume pointwise
+        f, r, w = mfu_pointwise(inputs, output)
+        log = f"""Assuming {op_name} is pointwise, \
+with {f} flops, {r} read bytes, {w} written bytes"""
+        logger.warning(log)
+        return f, r, w
+    return 0, 0, 0
+
+
+def analyze_model_mfu_fast(model, parameter_count, args, kwargs):
+    if 'input_ids' not in kwargs:
+        return MFUInfo()
+    assert parameter_count > 0, "Please specify mfu_analysis_active_parameters"
+    mfu_flops = kwargs['input_ids'].numel() * parameter_count * 2
+    return MFUInfo(mfu_flops)
+
+
+def analyze_model_mfu(model, args, kwargs):
+
+    ep = export(model, args=args, kwargs=kwargs)
+
+    mfu_info = MFUInfo()
+    for node in ep.graph.nodes:
+        f, r, w = analyze_node(node, model)
+        mfu_info.flops += f
+        mfu_info.read_bytes += r
+        mfu_info.write_bytes += w
+    return mfu_info
+
+
+class MFUAnalysisLogging:
```

Comment:
Makes sense to me to put this class under `vllm/v1/metrics`, and any code that is exclusively used by this class